File: programmers/js/basic/181832.js
// 정수를 나선형으로 배치하기
function solution(n) {
  const arr = Array.from({ length: n }, () => Array(n).fill(0));
  let i = 0;
  let j = 0;
  let num = 1;
  const progress = () => {
    arr[i][j] = num;
    num++;
  };

  progress();

  while (num <= n * n) {
    while (j + 1 < n && arr[i][j + 1] === 0) {
      j++;
      progress();
    }

    while (i + 1 < n && arr[i + 1][j] === 0) {
      i++;
      progress();
    }

    while (j - 1 >= 0 && arr[i][j - 1] === 0) {
      j--;
      progress();
    }

    while (i - 1 >= 0 && arr[i - 1][j] === 0) {
      i--;
      progress();
    }
  }

  return arr;
}

File: programmers/java/basic/181832.java
// 정수를 나선형으로 배치하기

class Solution {
    public int[][] solution(int n) {
        int[][] result = new int[n][n];
        int row = 0;
        int col = 0;
        int num = 2;
        
        result[0][0] = 1;
        
        while (num <= n * n) {
            while (row + 1 < n && result[col][row + 1] == 0) {
                row += 1;
                result[col][row] = num;
                num += 1;
            }
            
            while (col + 1 < n && result[col + 1][row] == 0) {
                col += 1;
                result[col][row] = num;
                num += 1;
            }
            
            while (row - 1 >= 0 && result[col][row - 1] == 0) {
                row -= 1;
                result[col][row] = num;
                num += 1;
            }
            
            while (col - 1 >= 0 && result[col - 1][row] == 0) {
                col -= 1;
                result[col][row] = num;
                num += 1;
            }
        }
        
        
        return result;
    }
}